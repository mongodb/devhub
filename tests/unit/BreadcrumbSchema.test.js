import React from 'react';
import { shallow } from 'enzyme';
import BreadcrumbSchema from '../../src/components/dev-hub/breadcrumb-schema';

import mockData from './data/BreadcrumbSchema.test.json';

const siteUrl = 'https://developer.mongodb.com';

jest.mock('../../src/hooks/use-site-metadata', () => ({
    useSiteMetadata: () => ({ siteUrl })
}));

describe('BreadcrumbSchema', () => {
    let shallowWrapper;

    beforeAll(() => {
        shallowWrapper = shallow(<BreadcrumbSchema breadcrumb={mockData} />);
    });

    it('renders correctly', () => {
        console.log(shallowWrapper);
        expect(shallowWrapper).toMatchSnapshot();
    });

    it('script has a correct schema', () => {
        const script = shallowWrapper.find('script');
        
        expect(script.text()).toEqual(JSON.stringify(
            {
                "@context": "https://schema.org",
                "@type": "BreadcrumbList",
                itemListElement: [
                    { "@type": "ListItem", "position": 1, name: "Home", item: siteUrl }, 
                    { "@type": "ListItem", "position": 2, name: "Learn", item: siteUrl + "/learn/" }]
            }
        ));
    });
})
