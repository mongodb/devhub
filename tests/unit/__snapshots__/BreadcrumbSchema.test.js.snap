// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbSchema renders correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Home","item":"https://developer.mongodb.com"},{"@type":"ListItem","position":2,"name":"Learn","item":"https://developer.mongodb.com/learn/"}]}
  </script>
</HelmetWrapper>
`;
